### ISSUE TITLE
Error inflating ArSceneView on Pixel 2 XL

### ISSUE THREAD

**Issue by carterhudson** [google-ar/sceneform-android-sdk#188]
EDIT: I got it running. The portion of the tutorial for adding ARCore & requesting permissions is easy to miss and also out of date.

I can run the sceneform sample app on this device without a problem. My device shows its OpenGL version as 3.2. I get this when I try to put an ArFragment in my main activity's layout. 

```
 java.lang.RuntimeException: Unable to start activity ComponentInfo{com.carterhudson.dtwenty/com.carterhudson.dtwenty.MainActivity}: android.view.InflateException: Binary XML file line #6: Binary XML file line #22: Binary XML file line #22: Error inflating class com.google.ar.sceneform.ArSceneView
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: android.view.InflateException: Binary XML file line #6: Binary XML file line #22: Binary XML file line #22: Error inflating class com.google.ar.sceneform.ArSceneView
     Caused by: android.view.InflateException: Binary XML file line #22: Binary XML file line #22: Error inflating class com.google.ar.sceneform.ArSceneView
     Caused by: android.view.InflateException: Binary XML file line #22: Error inflating class com.google.ar.sceneform.ArSceneView
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:334)
        at android.view.LayoutInflater.createView(LayoutInflater.java:647)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at com.google.ar.sceneform.ux.BaseArFragment.onCreateView(BaseArFragment.java:153)
        at android.support.v4.app.Fragment.performCreateView(Fragment.java:2346)
        at android.support.v4.app.FragmentManagerImpl.ensureInflatedFragmentView(FragmentManager.java:1664)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1399)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1659)
        at android.support.v4.app.FragmentManagerImpl.addFragment(FragmentManager.java:1905)
        at android.support.v4.app.FragmentManagerImpl.onCreateView(FragmentManager.java:3715)
        at android.support.v4.app.FragmentController.onCreateView(FragmentController.java:114)
        at android.support.v4.app.FragmentActivity.dispatchFragmentsOnCreateView(FragmentActivity.java:374)
        at android.support.v4.app.BaseFragmentActivityApi14.onCreateView(BaseFragmentActivityApi14.java:39)
        at android.support.v4.app.FragmentActivity.onCreateView(FragmentActivity.java:68)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:780)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
        at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287)
        at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139)
        at com.carterhudson.dtwenty.MainActivity.onCreate(MainActivity.kt:17)
        at android.app.Activity.performCreate(Activity.java:7009)
        at android.app.Activity.performCreate(Activity.java:7000)
07-11 12:04:51.878 4633-4633/com.carterhudson.dtwenty E/AndroidRuntime:     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.BootstrapMethodError: Exception from call site #36 bootstrap method
        at com.google.ar.sceneform.rendering.LightProbe.loadInBackground(LightProbe.java:196)
        at com.google.ar.sceneform.rendering.LightProbe.access$400(LightProbe.java:37)
        at com.google.ar.sceneform.rendering.LightProbe$Builder.build(LightProbe.java:362)
        at com.google.ar.sceneform.Scene.setupLightProbe(Unknown Source:25)
        at com.google.ar.sceneform.Scene.<init>(Unknown Source:51)
        at com.google.ar.sceneform.SceneView.initialize(Unknown Source:49)
        at com.google.ar.sceneform.SceneView.<init>(Unknown Source:39)
        at com.google.ar.sceneform.ArSceneView.<init>(Unknown Source:0)
        	... 43 more
     Caused by: java.lang.NoClassDefFoundError: Invalid descriptor: chronometer.
        	... 51 more
```

**Comment by eugenelozada** [comment]
did you add Java 8 compatibility in your app gradle file?

compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

**Comment by AxesandGrinds** [comment]
I believe that that is already in the app gradle file. I am getting the same errors and it seems it is an OpenGL ES problem wherein adroid studio only goes up to OpenGL 2 instead of 4.6. Is there an update on fixing this issue?

**Comment by malik-at-work** [comment]
@carterhudson I'd be interested in updating any out of date documentation.  Could you point me to the problematic doc/section ?

**Comment by malik-at-work** [comment]
@AxesandGrinds we're tracking OpenGL ES Version requirement at  https://github.com/google-ar/sceneform-android-sdk/issues/4 but we are lowering the requirement from 3.1 to 3.0 in an upcoming release.

**Comment by AxesandGrinds** [comment]
@malik-at-work , do you mean in the sceneform1.4 release or a release after that? When can we expect the next release?

**Comment by malik-at-work** [comment]
I can't give specific dates.  This is expected imminently, but will go through our normal release process and can be delayed for dozens of reasons.

**Comment by AxesandGrinds** [comment]
Imminently would be amazing. As soon as Sceneform is working correctly, a new world of AR opens up for android. It's bound to be epic. We're just waiting on you guys now. Thanks for this knowledge.

**Comment by jwmcglynn** [comment]
Hey @AxesandGrinds, Sceneform 1.4 is now available which supports OpenGL ES 3.0.  Can you give the new version a try?

**Comment by jwmcglynn** [comment]
Closing this issue, please reopen or create a new issue if you're still encountering it with Sceneform 1.4.

**Comment by ryan-hodgman** [comment]
Notably, if I checkout an older version of the codebase where the `ArSceneView` was being inflated without issue and try to run the app it is now crashing!

A couple of odd observations:

- The AndroidX library has no impact upon the crash, with the crash still occurring with the `android.support.v4.app` library
- The crash is occurring on old versions of the codebase that were previously working (I triple-checked a few different versions in order to be 100% sure), even after uninstalling and reinstalling a newly built .apk
- I can sync the gradle files successfully, but attempting to clean or rebuild the project results in a compile-time failure while searching for a generated gradle task (`compileAsset-src_47main_47assets_47scene_95air`) that relates to an asset that doesn't exist in the older versions of the codebase! (IDE screenshot attached)
- Running ` ./gradlew build` through the command line fixes the compile-time failure upon rebuilding / cleaning through Android Studio, but a freshly deployed .apk (even one generated through `./gradlew assembleDebug` on the command line) exhibits the same crash log
- Restarting Android Studio has no effect on the crash
- Proguard notes an issue that looks related...
`Note: kotlin.jvm.internal.Reflection: can't find dynamically referenced class kotlin.reflect.jvm.internal.ReflectionFactoryImpl`
...however the app crashes even on builds where Proguard hasn't been run

What is going on!? 😕 

<img width="1402" alt="screen shot 2018-11-19 at 1 43 37 pm" src="https://user-images.githubusercontent.com/26558824/48683471-811bb800-ec01-11e8-8c6f-6e35b405de4e.png">

**Comment by ryan-hodgman** [comment]
Debugging reveals that neither of the `ArSceneView`s constructors are called before the crash occurs. Could the issue be because `ArSceneView` lacks a zero-argument constructor to be called by `java.lang.reflect.Constructor.newInstance0(Native Method)`? I'm not sure why it would have been working on previous builds in that case.

**Comment by ryan-hodgman** [comment]
Alright, adding in the compatibility compile options appears to have resolved the issue...

```
android {
    // Sceneform libraries use language constructs from Java 8.
    // Add these compile options if targeting minSdkVersion < 26.
    compileOptions {
        sourceCompatibility 1.8
        targetCompatibility 1.8
    }
}
```

It might be worth updating the documentation on the Sceneform Quickstart page (https://developers.google.com/ar/develop/java/quickstart), since I had set `minSdkVersion 26` and the documentation led me to believe those options were unnecessary.

Also, I wonder if a more intuitive error could be caught and rethrown in this instance? I'm guessing people are getting stuck on this from time to time.

**Comment by chnouman** [comment]
Facing the same issue and didn't get resolved using compileOptions.
